src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -102,10 +173,19 @@ public void clearCacheForInstanceSource(ExternalDataInstanceSource source) throw
         String cacheKey = getCacheKey(source.getSourceUri(), source.getRequestData());
         Cache cache = cacheManager.getCache("case_search");
         cache.evict(cacheKey);
+
+        CaseSearchDB caseSearchDb = initCaseSearchDB();
+        String caseSearchTableName = evalCaseSearchTableName(cacheKey);
+        UserSqlSandbox caseSearchSandbox = new CaseSearchSqlSandbox(caseSearchTableName, caseSearchDb);
+        IStorageUtilityIndexed<Case> caseSearchStorage = caseSearchSandbox.getCaseStorage();
+        caseSearchStorage.deleteStorage();
```

Comment:
is it worth checking `caseSearchStorage.isStorageExists()` first?

Reply:
the delete does a `DROP TABLE IF EXISTS ` so it's not required to do a preliminary check.